### ISSUE TITLE
Any way to create ImageRGB object from Buffer ?

### ISSUE THREAD

**Issue by bobmoff** [justadudewhohacks/face-recognition.js#21]
Hi!

This is more of a question. Trying to figure out how to create ImageRGB object without reading from disk, but cant figure out what an ImageRGB really is. 

Any idea of how I could make the FaceRecognizer.AddFaces function support a buffer, or how to make an ImageRGB from a buffer or stream. Would be nice to not have to write stuff to disk, just to read it up again.

**Comment by bobmoff** [comment]
Looking at the javascript object returned from loadImage it contains on { cols, rows }, so my guess is that it some kind of pointer to the data somewhere else?

**Comment by justadudewhohacks** [comment]
Hi,

ImageRGB is a native class, just a wrapper for a dlib matrix with rgb pixel type. As far as I am aware there is no way in dlib to create such a matrix from raw data. If you wanted to do so you would have to create your own interface, which I did to create an opencv4nodejs interface: 
https://github.com/justadudewhohacks/face-recognition.js/blob/master/cc/GenericImage.h
https://github.com/justadudewhohacks/face-recognition.js/blob/master/cc/GenericImageBase.h

The question is, what kind of buffer do you have. Is it raw data with some pixel ordering e.g. rgb, bgr, rgba or is it a base64 encoded string...

Long story short, if you have an image source other then jpg or png images on your disk, such as Video, Webcam images, an image stream, raw image data, base64 encoded data, then I would highly suggest you to use face-recognition.js with [opencv4nodejs](https://github.com/justadudewhohacks/opencv4nodejs/). There are examples how to use this package with opencv4nodejs.

**Comment by bobmoff** [comment]
Thx!

**Comment by aesyungan** [comment]
what is the solution ?  I have an image in base64
